### ISSUE TITLE
Node server don't close connection after get browser lib

### ISSUE THREAD

**Issue by bato3** [faye/faye#468]
The browser library maintains a persistent connection to the server, (which kills my machine).
The connection is opened minimum until close tab.
If it was open several tabs - to close the browser.

When I get `http://localhost/faye/faye-browser-min.js` connection isn't close. 
`netstat` show me status: `ESTABLISHED` 

node6, faye 1.2.4, (firefox x64 | chrome)

**Comment by jcoglan** [comment]
Could you give me a script showing exactly how you're configuring your server?

All the paths in https://github.com/faye/faye/blob/master/src/adapters/static_server.js call `response.end()` and they don't set the `keep-alive` header, so the connection should be closed.

However this might be happening for some other reason -- if you can give me instructions for how to observe this happening, then I'll investigate.

**Comment by bato3** [comment]
My script is very simple: http://pastebin.com/LTXRDhUc

open in browser http://127.0.0.1/czat/czat.js or http://localhost/czat/czat.js

`netstat -an | grep 4201` after some time shows stats (2 lines), 1 listen server and extra ESTABLISHED connection.

Pervious I test on FreBSD, now on Debian

**Comment by bato3** [comment]
I found it!
When file is served with http 200 - it's OK, but 304 sends "keep-alive"

**Comment by jcoglan** [comment]
When I test this on Node 6.9.5, it sends `keep-alive` regardless of whether it's a 200 or 304 response. That header is not actually set by Faye, it's set by Node; the `http` module sends `Connection: keep-alive` by default.

Enabling keep-alive actually seems like a reasonable behaviour here; could you explain some more what problem it's causing?

**Comment by bato3** [comment]
@jcoglan: When I get http://localhost/faye/faye-browser-min.js connection isn't close until  client close browser / browser tab.

I'm using faye as push server for chat displayed for logged users. But I add <script src="http://localhost/faye/faye-browser-min.js"> for all clients.  Who needs and don't need push server. Average time on page is 30 minutes.  It takes me 20k (or more, I don't know, because I'm count ws clients not TCP connection) active connections. 
Finally I reached the limit of open files.

Now I don't include faye-browser-min.js for unlogged users (and logged have 2+ tabs open) so I don't have this problem now.

But I bought 2 extra servers for load balancing, and my webpage was unavailable over month. (loading over 1 sec, timeouts...)

**Comment by jcoglan** [comment]
Apologies that I've not had time recently to catch up on issues here. I've looked into this a little more and it seems like reasonably common behaviour; for example I tried fetching one of the scripts that GitHub is currently serving to me and it kept the connection alive to 10 minutes.

If this is a serious issue for you, I recommend configuring your reverse proxy in production to disable keep-alive on requests to Faye's static file paths.
